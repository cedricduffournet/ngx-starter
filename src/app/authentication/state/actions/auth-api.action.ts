import { props, createAction } from '@ngrx/store';

import { User } from '@app/user/models/User';
import { AuthToken } from '@app/authentication/models/auth';

export const loginSuccess = createAction(
  '[Auth/API] Login success',
  props<{ authToken: AuthToken }>()
);

export const loginFailure = createAction(
  '[Auth/API] Login failure',
  props<{ error: any }>()
);

export const registerSuccess = createAction(
  '[Auth/API] Register success',
  props<{ userRegistered: User }>()
);

export const registerFailure = createAction(
  '[Auth/API] Register failure',
  props<{ error: any }>()
);

export const loadLoggedUserSuccess = createAction(
  '[Auth/API] Load logged user success',
  props<{ user: User }>()
);

export const loadLoggedUserFailure = createAction(
  '[Auth/API] Load logged user failure',
  props<{ error: any }>()
);

export const loadTokenSuccess = createAction(
  '[Auth/API] load token success',
  props<{ authToken: AuthToken }>()
);

export const loadTokenFailure = createAction(
  '[Auth/API] load token failure',
  props<{ error: string }>()
);

export const logoutSuccess = createAction('[Auth] log out success');
