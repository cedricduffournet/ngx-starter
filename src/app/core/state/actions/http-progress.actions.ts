import { createAction, props } from '@ngrx/store';

export const begin = createAction(
  '[HTTP progress] Begin http call',
  props<{ url: string }>()
);

export const complete = createAction(
  '[HTTP complete] Complete http call',
  props<{ url: string }>()
);
