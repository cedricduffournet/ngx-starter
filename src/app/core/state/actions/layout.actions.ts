import { createAction } from '@ngrx/store';

export const toggleMenu = createAction('[Layout Core] toggle menu');
