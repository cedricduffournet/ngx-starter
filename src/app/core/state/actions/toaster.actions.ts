import { props, createAction } from '@ngrx/store';
import { ToasterParams } from '@app/core/containers/toaster';
export const pop = createAction(
  '[Toaster] Pop toaster',
  props<{ params: ToasterParams }>()
);
