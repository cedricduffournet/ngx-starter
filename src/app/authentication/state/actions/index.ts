import * as AuthActions from './auth.actions';
import * as AuthApiActions from './auth-api.action';
import * as LoginViewActions from './login-view.actions';
import * as RegisterViewActions from './register-view.actions';

export { AuthActions, AuthApiActions, LoginViewActions, RegisterViewActions };
