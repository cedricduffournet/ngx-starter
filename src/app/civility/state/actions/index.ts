import * as CivilityUpdateModalActions from './civility-update-modal.actions';
import * as CivilityAddModalActions from './civility-add-modal.actions';
import * as CivilityDeleteModalActions from './civility-delete-modal.actions';
import * as CivilityListViewActions from './civility-list-view.actions';
import * as CivilityApiActions from './civility-api.actions';

export {
  CivilityListViewActions,
  CivilityApiActions,
  CivilityUpdateModalActions,
  CivilityAddModalActions,
  CivilityDeleteModalActions
};
