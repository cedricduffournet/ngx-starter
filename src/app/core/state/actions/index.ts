import * as LayoutCoreActions from './layout.actions';
import * as ToasterActions from './toaster.actions';
import * as HttpProgressActions from './http-progress.actions';

export { LayoutCoreActions, ToasterActions, HttpProgressActions };
