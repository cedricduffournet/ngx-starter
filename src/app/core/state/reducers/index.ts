import { InjectionToken } from '@angular/core';
import {
  createFeatureSelector,
  createSelector,
  ActionReducerMap,
  Action
} from '@ngrx/store';
import * as fromRouter from '@ngrx/router-store';
import * as fromLayout from '@app/core/state/reducers/layout.reducer';
import * as fromToaster from '@app/core/state/reducers/toaster.reducer';
import * as fromHttpProgress from '@app/core/state/reducers/http-progress.reducer';
import * as fromAuth from '@app/authentication/state/reducers';
import { RouterStateUrl } from '@app/core/utils/custom-route-serializer';

export interface State {
  layout: fromLayout.State;
  toaster: fromToaster.State;
  http: fromHttpProgress.State;
  router: fromRouter.RouterReducerState<RouterStateUrl>;
}

export const reducers = new InjectionToken<ActionReducerMap<State, Action>>(
  'Root reducers token',
  {
    factory: () => ({
      layout: fromLayout.reducer,
      toaster: fromToaster.reducer,
      http: fromHttpProgress.reducer,
      router: fromRouter.routerReducer
    })
  }
);

export const getRouterState = createFeatureSelector<
  fromRouter.RouterReducerState<RouterStateUrl>
>('router');

export const getPageTitle = createSelector(
  getRouterState,
  router => {
    if (router.state.data && router.state.data.title) {
      return router.state.data.title;
    }
    return '';
  }
);

export const getPreviousLink = createSelector(
  getRouterState,
  state => {
    if (state.state.data && state.state.data.previousLink) {
      return state.state.data.previousLink;
    }
    return null;
  }
);

export const getLayoutState = createFeatureSelector<State, fromLayout.State>(
  'layout'
);

export const getShowSidenav = createSelector(
  getLayoutState,
  fromLayout.getShowSidenav
);

export const getMenu = createSelector(
  getLayoutState,
  fromLayout.getMenu
);

export const getUserMenu = createSelector(
  getMenu,
  fromAuth.getLoggedUser,
  (menu, loggedUser) => {
    if (!loggedUser) {
      return [];
    }
    return menu.filter(item =>
      item.roles.some(role => loggedUser.roles.indexOf(role) !== -1)
    );
  }
);

export const getToasterState = createFeatureSelector<State, fromToaster.State>(
  'toaster'
);

export const getToasterParams = createSelector(
  getToasterState,
  fromToaster.getParams
);

export const getHttpState = createFeatureSelector<
  State,
  fromHttpProgress.State
>('http');

export const getHttpProcessList = createSelector(
  getHttpState,
  fromHttpProgress.getProcessList
);

export const isHttpProcessing = createSelector(
  getHttpProcessList,
  processList => processList.length !== 0
);
