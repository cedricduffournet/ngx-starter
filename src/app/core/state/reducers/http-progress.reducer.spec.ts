import { reducer } from '@app/core/state/reducers/http-progress.reducer';
import * as fromHttpProgress from '@app/core/state/reducers/http-progress.reducer';

import { HttpProgressActions } from '@app/core/state/actions';

describe('HttpProgressReducer', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any);

      expect(result).toMatchSnapshot();
    });
  });

  describe('BEGIN', () => {
    it('should return state with processing true and new entry in proccess list and url', () => {
      const action = HttpProgressActions.begin({ url: '/test_begin' });
      const result = reducer(fromHttpProgress.INITIAL_STATE, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('COMPLETE', () => {
    it('should remove elem in process list, set url and set processing to false', () => {
      const initialState = {
        ...fromHttpProgress.INITIAL_STATE,
        processing: true,
        processList: [true]
      };
      const action = HttpProgressActions.complete({ url: '/test_complete' });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('Selectors', () => {
    describe('getProcessing', () => {
      it('shoud retrieve processing (true)', () => {
        const result = fromHttpProgress.getProcessing({
          ...fromHttpProgress.INITIAL_STATE,
          processing: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getProcessList', () => {
      it('shoud retrieve process list', () => {
        const result = fromHttpProgress.getProcessList({
          ...fromHttpProgress.INITIAL_STATE,
          processList: [true, true, true]
        });

        expect(result).toMatchSnapshot();
      });
    });
  });
});
