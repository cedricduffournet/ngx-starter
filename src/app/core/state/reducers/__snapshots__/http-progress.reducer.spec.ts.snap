// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HttpProgressReducer BEGIN should return state with processing true and new entry in proccess list and url 1`] = `
Object {
  "processList": Array [
    true,
  ],
  "processing": true,
  "url": "/test_begin",
}
`;

exports[`HttpProgressReducer COMPLETE should remove elem in process list, set url and set processing to false 1`] = `
Object {
  "processList": Array [],
  "processing": false,
  "url": "/test_complete",
}
`;

exports[`HttpProgressReducer Selectors getProcessList shoud retrieve process list 1`] = `
Array [
  true,
  true,
  true,
]
`;

exports[`HttpProgressReducer Selectors getProcessing shoud retrieve processing (true) 1`] = `true`;

exports[`HttpProgressReducer undefined action should return the default state 1`] = `
Object {
  "processList": Array [],
  "processing": false,
  "url": "",
}
`;
